File: Testio/app/src/main/java/com/azzdorfrobotics/android/testio/network/TesonetApi.java
package com.azzdorfrobotics.android.testio.network;

import com.azzdorfrobotics.android.testio.network.model.request.LoginRequest;
import com.azzdorfrobotics.android.testio.network.model.response.LoginResponse;
import com.azzdorfrobotics.android.testio.network.model.response.Server;
import io.reactivex.Observable;
import java.util.List;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;

/**
 * Created on 25.12.2017.
 *
 * @author Mykola Tychyna (iMykolaPro)
 */

public interface TesonetApi {

    String API_VERSION_1 = "/v1";

    @POST(API_VERSION_1 + "/tokens") Observable<LoginResponse>  login(@Body LoginRequest request);

    @GET(API_VERSION_1 + "/servers") Observable<List<Server>> servers(@Header("Authorization") String token);
}

File: app/src/main/kotlin/com/ne2rnas/party/network/TesonetApi.kt
package com.ne2rnas.party.network

import com.ne2rnas.party.data.login.LoginResponse
import com.ne2rnas.party.data.servers.Server
import io.reactivex.Observable
import retrofit2.http.*

interface TesonetApi {
    @FormUrlEncoded
    @POST("/v1/tokens")
    fun login(@Field("username") username: String,
              @Field("password") password: String): Observable<LoginResponse>

    @GET("/v1/servers")
    fun getServers(@Header("authorization") token: String?): Observable<List<Server>>
}
